### ISSUE TITLE
wildcard characters in regex now fail

### ISSUE THREAD

**Issue by lrstanley** [go-chi/chi#388]
I believe cd12532e9858f735c2cac6cc8a398dfd987676dd may have caused wildcard characters in regexes to trigger the `* must always be last` case, even though that shouldn't apply for things inside of a regex.

```go
r.Get("/{uri:[a-zA-Z0-9._-]+}/{user:[a-zA-Z0-9][-a-zA-Z0-9]+}/{repo:[a-zA-Z][-.a-zA-Z0-9]*}", func(w http.ResponseWriter, r *http.Request) {
		uri := chi.URLParam(r, "uri")
		user := chi.URLParam(r, "user")
		repo := chi.URLParam(r, "repo")
}
```

In this example, I'm using it to check for things like `/github.com/lrstanley/girc`. However, this fails, because it sees `*`, even though this is regex, and has nothing to do with chi globbing:

```
panic: chi: wildcard '*' must be the last value in a route. trim trailing text or use a '{param}' instead
```

**Comment by pkieltyka** [comment]
thanks for the report, will look into it

**Comment by pkieltyka** [comment]
fixed in 85905ae4dd0a60afa490c062913f0c3ae168ac7e
